src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/PetClusterServiceAccountProvider.scala

```diff
@@ -5,14 +5,14 @@ import org.broadinstitute.dsde.workbench.leonardo.model.ServiceAccountProvider
 import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail}
 import org.broadinstitute.dsde.workbench.model.google.GoogleProject
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent._
 
 class PetClusterServiceAccountProvider(val config: Config) extends ServiceAccountProvider(config) with SamProvider {
 
   override def getClusterServiceAccount(userInfo: UserInfo, googleProject: GoogleProject)(implicit executionContext: ExecutionContext): Future[Option[WorkbenchEmail]] = {
     // Ask Sam for a pet service account for the given (user, project)
     Future {
-      Option(samClient.getPetServiceAccount(userInfo, googleProject))
+      Option(blocking(samClient.getPetServiceAccount(userInfo, googleProject)))
```

Comment:
I also added `blocking` to all of our `SwaggerSamClient` calls, since they are indeed blocking.

Reply:
I have mixed feelings about blocking... doesn't it create a new thread just for that call, i.e. not pooled?

Reply:
Yeah it can create an extra thread, but I think that's probably better than blocking a pooled thread, since this is an IO-bound task.